main.go

```diff
@@ -41,10 +41,16 @@ func startLogServe(stream *stream.Stream) {
 		// URL and comparing the reminder to the accessToken, ensuring a URL pattern
 		// /log/<accessToken>
 		if r.URL.String()[5:] != accessToken {
-			w.Header().Set("Content-Type", "text/plain; charset=utf-8")
-			w.Header().Set("Access-Control-Allow-Origin", "*")
+			writeHeaders(w, r)
 			w.WriteHeader(401)
 			fmt.Fprint(w, "Access denied")
+		} else if r.Method == "HEAD" {
+			writeHeaders(w, r)
+			// If we are creating a HEAD request, we can also mark that the subsequent
+			// GET request exposes access to X-Streaming
+			w.Header().Set("Access-Control-Allow-Headers", "X-Streaming")
```

Comment:
That's not what this header does... It's presence in a pre-flight request (method: OPTIONS), allows subsequent requests to send the header.

Note: Pre-flight is only used for methods other than GET, HEAD, POST, or if certain headers are set, see:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Simple_requests

On simple requests vs. preflighted requests...
We only need simple requests (afaik), we're not sending special headers, only reading them.